platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/crypto/EnhancedKeyStoreLoader.java

```diff
@@ -1165,4 +1175,340 @@ private interface AddressBookCallback {
+            throws KeyStoreException, KeyLoadingException {
+        final KeyStore legacyPublicStore = resolveLegacyPublicStore();
+        AtomicLong errorCount = new AtomicLong(0);
+
+        iterateAddressBook(addressBook, (i, nodeId, address, nodeAlias) -> {
+            if (isLocal(address)) {
+                // extract private keys for local nodes
+                final Path sPrivateKeyLocation = keyStoreDirectory.resolve("s-private-" + nodeAlias + ".pem");
+                final Path ksLocation = legacyPrivateKeyStore(nodeAlias);
+                if (!Files.exists(sPrivateKeyLocation) && Files.exists(ksLocation)) {
+                    logger.trace(
+                            STARTUP.getMarker(),
+                            "Extracting private signing key for node {} from file {}",
+                            nodeId,
+                            ksLocation.getFileName());
+                    final PrivateKey privateKey =
+                            readLegacyPrivateKey(nodeId, ksLocation, KeyCertPurpose.SIGNING.storeName(nodeAlias));
+                    pfxPrivateKeys.put(nodeId, privateKey);
+                    if (privateKey == null) {
+                        logger.error(
+                                ERROR.getMarker(),
+                                "Failed to extract private signing key for node {} from file {}",
+                                nodeId,
+                                ksLocation.getFileName());
+                        errorCount.incrementAndGet();
+                    } else {
+                        logger.trace(
+                                STARTUP.getMarker(),
+                                "Writing private signing key for node {} to PEM file {}",
+                                nodeId,
+                                sPrivateKeyLocation.getFileName());
+                        try {
+                            writePemFile(true, sPrivateKeyLocation, privateKey.getEncoded());
+                        } catch (IOException e) {
+                            logger.error(
+                                    ERROR.getMarker(),
+                                    "Failed to write private key for node {} to PEM file {}",
+                                    nodeId,
+                                    sPrivateKeyLocation.getFileName());
+                            errorCount.incrementAndGet();
+                        }
+                    }
+                }
+            }
+
+            // extract certificates for all nodes
+            final Path sCertificateLocation = keyStoreDirectory.resolve("s-public-" + nodeAlias + ".pem");
+            Path ksLocation = legacyCertificateStore();
+            if (!Files.exists(sCertificateLocation) && Files.exists(ksLocation)) {
+                logger.trace(
+                        STARTUP.getMarker(),
+                        "Extracting signing certificate for node {} from file {} ]",
```

Comment:
is the trailing `]` in this message intentional?

Reply:
nope, removed.